testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveCompatibility.java

```diff
@@ -169,6 +171,47 @@ else if (!(isParquetStorageFormat && isParquetOptimizedWriterEnabled)) {
         onTrino().executeQuery(format("DROP TABLE %s", tableName));
     }
 
+    @Test(groups = STORAGE_FORMATS_DETAILED)
+    public void testTimestampFieldWrittenByOptimizedParquetWriterCannotBeReadByHive()
+            throws Exception
+    {
+        // only admin user is allowed to change session properties
+        setAdminRole(onTrino().getConnection());
+        setSessionProperty(onTrino().getConnection(), "hive.experimental_parquet_optimized_writer_enabled", "true");
+
+        String tableName = "parquet_table_timestamp_created_in_trino";
+        onTrino().executeQuery("DROP TABLE IF EXISTS " + tableName);
+        onTrino().executeQuery("CREATE TABLE " + tableName + "(timestamp_precision varchar, a_timestamp timestamp) WITH (format = 'PARQUET')");
+        for (HiveTimestampPrecision hiveTimestampPrecision : HiveTimestampPrecision.values()) {
+            setSessionProperty(onTrino().getConnection(), "hive.timestamp_precision", hiveTimestampPrecision.name());
+            onTrino().executeQuery("INSERT INTO " + tableName + " VALUES ('" + hiveTimestampPrecision.name() + "', TIMESTAMP '2021-01-05 12:01:00.111901001')");
+            // Hive expects `INT96` (deprecated on Parquet) for timestamp values
+            assertQueryFailure(() -> onHive().executeQuery("SELECT a_timestamp FROM " + tableName + " WHERE timestamp_precision = '" + hiveTimestampPrecision.name() + "'"))
+                    .hasMessageMatching(".*java.lang.ClassCastException: org.apache.hadoop.io.LongWritable cannot be cast to org.apache.hadoop.hive.serde2.io.(TimestampWritable|TimestampWritableV2)");
+        }
+        onTrino().executeQuery(format("DROP TABLE %s", tableName));
+    }
+
+    @Test(groups = STORAGE_FORMATS_DETAILED)
+    public void testSmallDecimalFieldWrittenByOptimizedParquetWriterCannotBeReadByHive()
+            throws Exception
+    {
+        // only admin user is allowed to change session properties
+        setAdminRole(onTrino().getConnection());
+        setSessionProperty(onTrino().getConnection(), "hive.experimental_parquet_optimized_writer_enabled", "true");
+
+        String tableName = "parquet_table_small_decimal_created_in_trino";
+        onTrino().executeQuery("DROP TABLE IF EXISTS " + tableName);
+        onTrino().executeQuery("CREATE TABLE " + tableName + " (a_decimal DECIMAL(5,0)) WITH (format='PARQUET')");
+        onTrino().executeQuery("INSERT INTO " + tableName + " VALUES (123)");
+
+        // Hive expects `FIXED_LEN_BYTE_ARRAY` for decimal values irrespective of the Parquet specification which allows `INT32`, `INT64` for short precision decimal types
+        assertQueryFailure(() -> onHive().executeQuery("SELECT a_decimal FROM " + tableName))
+                .hasMessageMatching(".* org.apache.parquet.io.ParquetDecodingException: Can not read value at 1 in block 0 in file .*");
```

Comment:
I think this is consistently failing with missing `org.apache.` in the package name:
```
tests               | 2022-01-31 12:13:37 INFO: FAILURE     /    io.trino.tests.product.hive.TestHiveCompatibility.testSmallDecimalFieldWrittenByOptimizedParquetWriterCannotBeReadByHive (Groups: storage_formats_detailed) took 0.7 seconds
tests               | 2022-01-31 12:13:37 SEVERE: Failure cause:
tests               | java.lang.AssertionError: 
tests               | Expecting message:
tests               |   <"java.io.IOException: parquet.io.ParquetDecodingException: Can not read value at 1 in block 0 in file hdfs://hadoop-master:9000/user/hive/warehouse/parquet_table_small_decimal_created_in_trino/20220131_062836_01082_j2ikr_8c200c2a-4add-4f35-a3ed-b70c4732e54b">
tests               | to match regex:
tests               |   <".* org.apache.parquet.io.ParquetDecodingException: Can not read value at 1 in block 0 in file .*">
tests               | but did not.
```

Reply:
Example: https://github.com/trinodb/trino/runs/5001778522?check_suite_focus=true

Reply:
fixed now